h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -287,16 +305,47 @@ public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
     return performGroupingWithAggregations(fr, gbCols, aggs);
   }
 
+  /**
+   * Performs grouping by numeric columns and aggregations on given data.
+   * 
+   * @param fr      frame on which grouping should be performed
+   * @param gbCols  indices of numeric columns (of given frame) to be used for grouping.
+   *                For string columns see:
+   *                {@link AstGroup#performGroupingWithAggregations(Frame, int[], GroupByColType[], AGG[])}.
+   * @param aggs    aggregations to be performed on the grouped data.
+   * 
+   * @return  grouped data with aggregations results.
+   */
   public ValFrame performGroupingWithAggregations(Frame fr, int[] gbCols, AGG[] aggs) {
+    GroupByColType[] gbColsTypes = Collections.nCopies(gbCols.length, GroupByColType.NUMERIC)
+                                    .toArray(new GroupByColType[gbCols.length]);
+    return performGroupingWithAggregations(fr, gbCols, gbColsTypes, aggs);
+  }
+
+  /**
+   * Performs grouping and aggregations on given data.
+   *
+   * @param fr          frame on which grouping should be performed
+   * @param gbCols      indices of columns (of given frame) to be used for grouping.
+   * @param gbColsTypes types of columns which are being used for grouping.
+   * @param aggs        aggregations to be performed on the grouped data.
+   *
+   * @return  grouped data with aggregations results.
+   */
+  public ValFrame performGroupingWithAggregations(Frame fr, int[] gbCols, GroupByColType[] gbColsTypes, AGG[] aggs) {
```

Comment:
each vec has type, why do we need to pass the col types? when the method can figure it out?

Reply:
True. Careless mistake. Thank you. I changed it.